### ISSUE TITLE
Default choice for search path in VS Code could be improved

### ISSUE THREAD

**Issue by gramster** [microsoft/pyright#294]
I've opened a project in VS Code in a "scratchpad" directory. I have no pyright config file.

On startup I see this:

```
Pyright language server starting
Starting service instance "scratchpad"
Starting service instance "<default>"
Fetching settings for workspace(s)
Received updated settings
No configuration file found.
Setting pythonPath for service "scratchpad": "/Users/grwheele/repos/scratchpad/python"
Searching for source files
No source files found.
Setting pythonPath for service "<default>": "/Users/grwheele/repos/scratchpad/python"
Searching for source files
No source files found.
```

There is no `python` subdirectory of `scratchpad` so this seems like a poor choice. Why not just use the root of the workspace?

**Comment by erictraut** [comment]
Pyright looks for the VS Code setting "python.pythonpath". If this setting is provided, it interprets it as a path. If it's an absolute path, it uses that. If it's a relative path, it assumes that it's relative to the root of your workspace. Does that behavior sound correct to you, or should pyright be treating this setting differently?

Apparently, in your case the "python.pythonpath" setting contains the string "python", which is being interpreted as a relative path. I just checked on my system, and the workspace setting for "python.pythonpath" is also set to "python". I don't remember setting it to this value, so it must be a default that's provided by the VS Code Python extension.

**Comment by DonJayamanne** [comment]
> Apparently, in your case the "python.pythonpath" setting contains the string "python", which is being interpreted as a relative path. I just

The value `python` is special.
We treat this as the default python executable in the current path. I.e. when you open a terminal and type `python`. That is what this value refers to.
Otherwise it's a relative path.

**Here's the logic:**
* If value is `python`, then use system default python interpreter in current path
* If value is absolute then use that as is
* If value is relative, then it's relative to the workspace directory

**Comment by erictraut** [comment]
Thanks for the clarification! I'll fix the logic in pyright to match.

**Comment by erictraut** [comment]
This change will be included in the next published version of pyright.

**Comment by erictraut** [comment]
This is now fixed in 1.0.80, which I just published.
